docs/developer/decision-records/2022-09-18-ids-requests-filtering/README.md

```diff
@@ -0,0 +1,42 @@
+# Add filtering to IDS requests
+
+## Decision
+
+Extend existing `CatalogRequest` object with list of criteria used to filter `Assets` within `Catalog` response.
+
+## Rationale
+
+In a large scale environment, with multiple EDCs holding the information about hundreds of thousands of `Assets`, instead of 
+looping thought all of the `ContractOffers`, there is a need to narrow the results down to even single offer. For example, and 
+`Asset` might hold the 'special' type of data (like registry or database) and should be searchable by its type.
+
+## Approach
```

Comment:
IMO this section can be shortened because only two things need to be done:
- replace the use of `Range` (for paging) with `QuerySpec`, for full querying capabilities
- swap out the (flattened) serialization of the `Range` with object serialization of the `QuerySpec`

Reply:
Done.

Reply:
> Done.

nope, it's not. Please highlight that transmitting the `Range` needs to be replaced by transmitting the entire `QuerySpec`. Filtering only is one aspect of it. Sorting is another.

Reply:
@paullatzelsperger what do you think about @ndr-brt reasoning? I agree with such argument.

Reply:
@ndr-brt @maciejkizlich-zf here's a weird thing: `ContractOfferQuery.getCriteria()`, `.getOffset()` and `.getLimit()` appear not to be used anywhere, at least IntelliJ doesn't report any findings.
Could we just use those fields? We might even be able to remove the `range` parameter from the `ContractOfferServiceImpl.queryContractOffers` then.

Reply:
@maciejkizlich-zf could you adapt your PR as per my last comment regarding the `ContractOfferQuery`?

Reply:
@paullatzelsperger please take a look now, I've added a part regarding ContractOfferQuery. 

And yes, we should be able to remove Range from the queryContractOffers(). Also, I'd replace .getOffset and getLimit() with just getRange().